xbmc/cores/VideoPlayer/DVDCodecs/Video/DXVA.cpp

```diff
@@ -1054,6 +1069,8 @@ void CDecoder::Close()
   {
     auto dxva_context = m_dxvaContext;
     CLog::LogF(LOGINFO, "closing decoder.");
+    if (m_avD3D11Context && m_avD3D11Context->surface)
+      m_dxvaContext->DestroySurfaces(m_avD3D11Context->surface_count, m_avD3D11Context->surface);
```

Comment:
buffer pool still keep the references to a surfaces.
call `m_bufferPool.reset()` is enought to release all of allocated surfaces.

Reply:
Yes, confirmed crash with current PR code due this:

![crash](https://user-images.githubusercontent.com/58434170/229350763-f4a6cf14-546f-47c4-a704-9b29982c5a99.png)

This exception not happens removing `m_dxvaContext->DestroySurfaces()`

Reply:
That's weird, I get that exception as well but only if I put a breakpoint in CVideoBufferPool::Reset.

Are there really multiple threads involved when closing?

I added a call to m_bufferPool.reset() earlier than its normal location but the views are still referenced and come up in ReportLiveDeviceObjects() output. I tried the case where a separate device is created for the decoder and the case where the main device is used => no difference
unless I don't understand that report well enough, will have to dig some more.

Reply:
I think is a race condition. Crash is real: on Xbox I reproduced full crash at video close with regular release build but not always (maybe 1 of 10 times closing video).

Reply:
It looks like that bufferPool is destroyed already. weird. if so then it seems like renderer still holds surface references. need to check this hypothese

Reply:
I think I found the reason and there is no leak so I'll withdraw that change.

basically the decoder is freed before the surfaces are released, so a report of objects when the decoder closes does not reflect the full picture.

buffer_pool.reset() with no effect in CDecoder::Close(): the buffer pool is kept alive by a buffer held for videoplayer
It eventually gets freed and releases the views.

I placed a call to ReportLiveDeviceObjects() in CVideoPlayerVideo::CloseStream() after m_picture.videoBuffer->Release(); and there are no more references to decoder output views so no leak.